beamer/chain.py

```diff
@@ -112,14 +116,22 @@ def __init__(self, context: Context):
         self._have_new_events = threading.Event()
         self._events: list[Event] = []
         self._stop = False
-        self._log = structlog.get_logger(type(self).__name__)
+        self._log = structlog.get_logger(type(self).__name__).bind(
+            source_chain_id=context.source_chain_id,
+            target_chain_id=context.target_chain_id,
+        )
         # The number of times we synced with a chain:
         # 0 = we're still waiting for sync to complete for chains
         # 1 = one of the chains was synced, if there is only one chain, sync done.
         # 2 = If there are 2 different chains, the sync is done
         self._num_syncs_done = 0
         self._context = context
         self._chain_ids = {self._context.source_chain_id, self._context.target_chain_id}
+        global log
+        log = log.bind(
+            source_chain_id=self._context.source_chain_id,
+            target_chain_id=self._context.target_chain_id,
+        )
```

Comment:
This won't work -- you're changing the global `log` object, which will not have the intended effect: all the event processors will use the same log object and therefore the same pair of chain IDs. This needs to be per-instance,
e.g.
```python
self._log = log.bind(...)
```

Additionally, could you please move these logging-related changes into their own, separate commit, to make the whole PR easier to review?

Reply:
This was aiming to affect all parts like proceess_request and process_claims that they are not part of event processor and if i dont have logger in context itself it wont work with bind